sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -86,7 +87,28 @@ export class EventHubProducer {
       throw error;
     }
 
-    let maxMessageSize = await this._eventHubSender!.getMaxMessageSize(options.abortSignal);
+    const retryOptions = this._senderOptions.retryOptions;
+    const maxRetries =
+      retryOptions && typeof retryOptions.maxRetries === "number"
+        ? retryOptions.maxRetries
+        : Constants.defaultMaxRetries;
+    const retryInterval =
+      retryOptions &&
+      typeof retryOptions.retryInterval === "number" &&
+      retryOptions.retryInterval > 0
+        ? retryOptions.retryInterval / 1000
+        : Constants.defaultDelayBetweenOperationRetriesInSeconds;
+
+    const config: RetryConfig<number> = {
+      operation: () => this._eventHubSender!.getMaxMessageSize(options!.abortSignal),
+      connectionId: this._context.connectionId,
+      operationType: RetryOperationType.senderLink,
```

Comment:
Hmm... Since the operation type is sender link... it is kinda misleading because we are passing `getMaxMessageSize` as the operation. So, earlier assessment that we should have the `retry()` being called here might not be a good idea.

We should probably move the `retry()` call inside `getMaxMessageSize` and pass `init()`  as "operation"

Reply:
Updated!!